Remove try with resource for Worfklow Worker Examples and use latest changes from durabletask-java

siri-varma opened the pull request:
# Description

_Please explain the changes you've made_

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #_[issue number]_

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Extended the documentation

siri-varma commented:
@cicoyle @salaboy @artur-ciocanu Can you folks please review this PR ? 

this is to incorporate the durabletask-java changes. Build will pass once we incorporate the new release.

I removed the autocloseable for WorkflowRuntime. Because with the new durable-task changes, once the close method is called we will stop accepting new requests.

cicoyle reviewed (dismissed):
thanks for adding the ternary and the followup here 👍🏻 lgtm. just need to wait for the dapr/durabletask-java release to show up in maven central and let CI run green then we can merge

artur-ciocanu commented on sdk-workflows/src/main/java/io/dapr/workflows/runtime/WorkflowRuntime.java:
```diff
@@ -50,11 +67,31 @@ public void start(boolean block) {
+    } catch (InterruptedException ex) {
+      Thread.currentThread().interrupt();
     }
   }
```
Please add a new line

artur-ciocanu commented:
@siri-varma and @cicoyle I really the like the idea of being able to provide an `ExecutorService` to GRPC worker, but I am not following why we need to get rid of `AutoClosable`.

@siri-varma could you please provide a little bit more details. Thank you.

artur-ciocanu reviewed (dismissed):
@siri-varma LGTM, I get it why we need `ExecutorService` changes, but I am not following why we need to remove `AutoClosable`

siri-varma commented:
@artur-ciocanu The durabletask-java SDK does not manage the gRPC channel when one is externally supplied. In our case, since the java-sdk provides the channel, we are responsible for managing its lifecycle.

As part of this PR, I’ve added close() and shutdown() methods in WorkflowRuntime to ensure proper cleanup.

As a result, in case of workers, since the gRPC server needs to run continuously, we cannot use try-with-resources here — doing so would automatically invoke close(), which would prematurely shut down the server.

---------------------------------------------------

Also, the AutoCloseable is still used in the WorkflowRuntime. Just that I removed it from the Worker examples because of the above reason.

siri-varma commented:
@artur-ciocanu The durabletask-java SDK does not manage the gRPC channel when one is externally supplied. In our case, since the java-sdk provides the channel, we are responsible for managing its lifecycle.

As part of this PR, I’ve added close() and shutdown() methods in WorkflowRuntime changes to ensure proper cleanup of the channel.

As a result, in case of workers, since the gRPC server needs to run continuously, we cannot use try-with-resources here — doing so would automatically invoke close(), which would prematurely shut down the server.

---------------------------------------------------

Also, the AutoCloseable is still being used in the Work

siri-varma commented:
@cicoyle I will pull in your changes once they are merged.

siri-varma commented:
Closing this. All the commits wen in this PR https://github.com/dapr/java-sdk/pull/1336
